processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -19,261 +19,78 @@
 import com.google.common.collect.Lists;
-import com.google.common.collect.Maps;
-import com.google.common.collect.Ordering;
 import com.google.common.collect.PeekingIterator;
 import com.google.common.collect.Sets;
-import com.google.common.io.ByteSink;
-import com.google.common.io.FileWriteMode;
-import com.google.common.io.Files;
-import com.google.common.io.OutputSupplier;
 import com.google.common.primitives.Ints;
-import com.google.common.primitives.Longs;
-import com.google.inject.Inject;
-import io.druid.collections.CombiningIterable;
-import io.druid.common.guava.FileOutputSupplier;
-import io.druid.common.guava.GuavaUtils;
-import io.druid.common.utils.JodaUtils;
+import com.google.inject.ImplementedBy;
 import io.druid.common.utils.SerializerUtils;
-import io.druid.java.util.common.IAE;
 import io.druid.java.util.common.ISE;
 import io.druid.java.util.common.Pair;
 import io.druid.java.util.common.guava.Comparators;
-import io.druid.java.util.common.guava.FunctionalIterable;
-import io.druid.java.util.common.guava.MergeIterable;
 import io.druid.java.util.common.guava.nary.BinaryFn;
-import io.druid.java.util.common.io.Closer;
-import io.druid.java.util.common.io.smoosh.Smoosh;
 import io.druid.java.util.common.logger.Logger;
 import io.druid.query.aggregation.AggregatorFactory;
-import io.druid.segment.column.ColumnCapabilities;
 import io.druid.segment.column.ColumnCapabilitiesImpl;
-import io.druid.segment.column.ValueType;
-import io.druid.segment.data.BitmapSerdeFactory;
-import io.druid.segment.data.CompressedObjectStrategy;
-import io.druid.segment.data.CompressionFactory;
-import io.druid.segment.data.GenericIndexed;
-import io.druid.segment.data.IOPeon;
 import io.druid.segment.data.Indexed;
-import io.druid.segment.data.LongSupplierSerializer;
-import io.druid.segment.data.TmpFileIOPeon;
 import io.druid.segment.incremental.IncrementalIndex;
-import io.druid.segment.incremental.IncrementalIndexAdapter;
-import io.druid.segment.serde.ComplexMetricColumnSerializer;
-import io.druid.segment.serde.ComplexMetricSerde;
-import io.druid.segment.serde.ComplexMetrics;
 import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
 import it.unimi.dsi.fastutil.ints.IntIterator;
 import it.unimi.dsi.fastutil.ints.IntSortedSet;
-import org.apache.commons.io.FileUtils;
-import org.joda.time.DateTime;
 import org.joda.time.Interval;
 
 import javax.annotation.Nullable;
-import java.io.Closeable;
 import java.io.File;
-import java.io.FileOutputStream;
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.nio.IntBuffer;
-import java.nio.channels.FileChannel;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Comparator;
 import java.util.Iterator;
 import java.util.List;
-import java.util.Map;
 import java.util.NoSuchElementException;
 import java.util.PriorityQueue;
 import java.util.Set;
+import java.util.stream.Collectors;
 
-/**
- */
-public class IndexMerger
+@ImplementedBy(IndexMergerV9.class)
```

Comment:
It is injected somewhere in indexing already. Other implementation is expected with segment format v10.